app/models/sms/incoming.rb

```diff
@@ -8,6 +8,14 @@ class Sms::Incoming < Sms::Message
     return true
   end
 
+  def self.search_qualifiers
+    super + [
+      Search::Qualifier.new(name: "user", col: "users.login", type: :text, assoc: :users, default: true),
+      Search::Qualifier.new(name: "name", col: "users.name", type: :text, assoc: :users, default: true),
+      Search::Qualifier.new(name: "number", col: "sms_messages.from", type: :text, default: true),
+    ]
+  end
```

Comment:
I see what you're doing here, but since search_qualifiers is always called on Sms::Message (since that's where do_search is called), I think these qualifiers will never get loaded. I will move them into Sms::Message.